cpp/arcticdb/column_store/segment_utils.hpp

```diff
@@ -8,32 +8,262 @@
 
 #pragma once
 
-#include <arcticdb/column_store/column.hpp>
-#include <arcticdb/column_store/column_algorithms.hpp>
 #include <ankerl/unordered_dense.h>
-#include <arcticdb/util/configs_map.hpp>
+#include <arcticdb/util/type_traits.hpp>
+#include <arcticdb/entity/types.hpp>
+#include <arcticdb/entity/stream_descriptor.hpp>
 #include <arcticdb/column_store/memory_segment.hpp>
+#include <arcticdb/pipeline/frame_slice.hpp>
+#include <arcticdb/pipeline/input_frame.hpp>
+#include <ranges>
+#include <fmt/format.h>
 
 namespace arcticdb {
 
-inline ankerl::unordered_dense::set<entity::position_t> unique_values_for_string_column(const Column& column) {
-    ankerl::unordered_dense::set<entity::position_t> output_set;
-    // Guessing that unique values is a third of the column length
-    // TODO would be useful to have actual unique count here from stats
-    static auto map_reserve_ratio = ConfigsMap::instance()->get_int("UniqueColumns.AllocationRatio", 3);
-    output_set.reserve(column.row_count() / map_reserve_ratio);
-
-    details::visit_type(column.type().data_type(), [&](auto col_desc_tag) {
-        using type_info = ScalarTypeInfo<decltype(col_desc_tag)>;
-        if constexpr (is_sequence_type(type_info::data_type)) {
-            arcticdb::for_each<typename type_info::TDT>(column, [&output_set](auto value) {
-                output_set.emplace(value);
-            });
```

Comment:
This is moved inside `cpp/arcticdb/column_store/segment_utils.cpp` without any changes